addons/mail/models/mail_activity.py

```diff
@@ -197,6 +196,26 @@ def _compute_state_from_date(self, date_deadline, tz=False):
         else:
             return 'planned'
 
+    @api.depends('res_model', 'res_id', 'user_id')
+    def _compute_can_write(self):
+        allowed = self.check_access_rights('write', raise_exception=False)
```

Comment:
I think that only valid_records = self._filter_access_rules('unlink') is usefull since filter_acess_rules will return an empty recordset if check_access_rights is false

def _compute_can_write(self):
    valid_records = self._filter_access_rules('write')
    for record in self:
            record.can_write = record in valid_records